### PR TITLE
klaviyo plugin support

### PR DESCRIPTION
**Pull request by poojarazor**

### CODE REVIEW

File: includes/api/save-abandonment-data.php [hunk 1]
```diff
@@ -63,6 +63,39 @@ function saveCartAbandonmentData(WP_REST_Request $request)
...
+
+        //Get token from kalviyo plugin
+        $klaviyoApi =  WooCommerceKlaviyo::instance();
+        $token = $klaviyoApi->options->get_klaviyo_option( 'klaviyo_public_api_key' );
+        $eventObject = ['token'=>$token,'event'=>'$started_checkout','customer_properties'=>array('$email'=>$email),'properties'=>$eventData];
+        $dataParam = json_encode($eventObject);
+        $data = base64_encode($dataParam);
+        $event = 'track';
+
+        $logObj['klaviyoData'] = $dataParam;
```
Comment (varun-315) [inline includes/api/save-abandonment-data.php:93]: make sure you are not logging their keys or PII into our system. Please strip the information if needed

Comment (poojarazor) [reply includes/api/save-abandonment-data.php:93]: got it

File: includes/api/save-abandonment-data.php [hunk 2]
```diff
@@ -63,6 +63,39 @@ function saveCartAbandonmentData(WP_REST_Request $request)
...
+        unset($eventData['Quantity']);
+        $email = $customerEmail;
+
+        //Get token from kalviyo plugin
+        $klaviyoApi =  WooCommerceKlaviyo::instance();
+        $token = $klaviyoApi->options->get_klaviyo_option( 'klaviyo_public_api_key' );
+        $eventObject = ['token'=>$token,'event'=>'$started_checkout','customer_properties'=>array('$email'=>$email),'properties'=>$eventData];
+        $dataParam = json_encode($eventObject);
+        $data = base64_encode($dataParam);
+        $event = 'track';
```
Comment (ChetanGN) [inline includes/api/save-abandonment-data.php:91]: @poojarazor Only track event is enough here? What about others? Will it not be required?

Comment (poojarazor) [reply includes/api/save-abandonment-data.php:91]: By this event we are able to trigger "started checkout" event

File: includes/api/save-abandonment-data.php [hunk 3]
```diff
@@ -63,6 +63,39 @@ function saveCartAbandonmentData(WP_REST_Request $request)
...
+
+        $logObj['klaviyoData'] = $eventData;
```
Comment (ChetanGN) [inline includes/api/save-abandonment-data.php:93]: @poojarazor Run formatter on this file.

**Review by ChetanGN — APPROVED** [review]
LGTM. Merge after code format.

**Pull request merged**